src/System.Text.Json/tests/JsonNumberTests.cs

```diff
@@ -0,0 +1,679 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Globalization;
+using Xunit;
+
+namespace System.Text.Json.Tests
+{
+    public static class JsonNumberTests
+    {
+        private delegate bool TryGetValue<T>(JsonNumber number, out T result);
+
+        private static void TestInitialization<T>(
+                T value,
+                Func<T, JsonNumber> ctor,
+                Action<JsonNumber, T> setter,
+                Func<JsonNumber, T> getter,
+                TryGetValue<T> tryGetter,
+                Func<T, JsonNumber> implicitCaster)
+        {
+            // Default constructor:
+            JsonNumber number = new JsonNumber();
+            setter(number, value);
+            AssertValue(value, number, getter, tryGetter);
+
+            // Numeric type constructor:
+            number = ctor(value);
+            AssertValue(value, number, getter, tryGetter);
+
+            // String constructor:
+            number = value switch
+            {
+                // Adding CultureInfo.InvariantCulture to support tests on plantforms where `,` is a  default decimal separator instead of `.`
```

Comment:
nit:
```suggestion
                // Adding CultureInfo.InvariantCulture to support tests on platforms where `,` is a  default decimal separator instead of `.`
```